src/test/java/fi/otavanopisto/kuntaapi/test/utils/PtvDownload.java

```diff
@@ -0,0 +1,218 @@
+  }
+  
+  private File getOrganizationsFolder() {
+    return new File(getPtvFolder(), "organizations");
+  }
+
+  private File getServicesFolder() {
+    return new File(getPtvFolder(), "services");
+  }
+
+  private File getServiceChannelsFolder() {
+    return new File(getPtvFolder(), "servicechannels");
+  }
+  
+  private File getPtvFolder() {
+    return new File(System.getProperty("user.dir"), "src/test/resources/ptv");
+  }
+  
+  private void downloadOrganizationServices(File organizationFile) throws IOException, JsonParseException, JsonMappingException {
+    V7VmOpenApiOrganization organization = getObjectMapper().readValue(organizationFile, V7VmOpenApiOrganization.class);
+    List<V5VmOpenApiOrganizationService> services = organization.getServices();
+    for (V5VmOpenApiOrganizationService service : services) {
+      downloadService(service);
+    }
+  }
+  
+  private ObjectMapper getObjectMapper() {
+    return new ObjectMapper();
+  }
+
+  private void downloadService(V5VmOpenApiOrganizationService service) throws JsonParseException, JsonMappingException, IOException {
+    String serviceId = service.getService().getId().toString();
+    File servicesFolder = getServicesFolder();
+    
+    String url = String.format("https://api.palvelutietovaranto.trn.suomi.fi/api/%s/Service/%s", PTV, serviceId);
+    File target = new File(servicesFolder, String.format("%s.json", serviceId));
+    if (!target.exists()) {
+      try {
+        download(url, target);          
+        System.out.println(url);
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+    }
+    
+    downloadServiceChannels(target);
+  }
+  
+  private void downloadServiceChannels(File serviceFile) throws JsonParseException, JsonMappingException, IOException {
+    V7VmOpenApiService service = new ObjectMapper().readValue(serviceFile, V7VmOpenApiService.class);
+    
+    List<V7VmOpenApiServiceServiceChannel> serviceChannels = service.getServiceChannels();
+    for (V7VmOpenApiServiceServiceChannel serviceChannel : serviceChannels) {
+      downloadServiceChannel(serviceChannel);
+    }
+  }
+
+  private void downloadServiceChannel(V7VmOpenApiServiceServiceChannel serviceChannel) {
+    String serviceChannelId = serviceChannel.getServiceChannel().getId().toString();
+    File serviceChannelsFolder = getServiceChannelsFolder();
+    
+    String url = String.format("https://api.palvelutietovaranto.trn.suomi.fi/api/%s/ServiceChannel/%s", PTV, serviceChannelId);
+    File target = new File(serviceChannelsFolder, String.format("%s.json", serviceChannelId));
+    if (!target.exists()) {
+      try {
+        download(url, target);          
+        System.out.println(url);
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+    } else {
+      System.out.println(target.getAbsolutePath());
+    }
+  }
+
+  private void download(String url, File target) throws IOException, MalformedURLException, FileNotFoundException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.net.MalformedURLException' which is a subclass of 'java.io.IOException'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.io.FileNotFoundException' which is a subclass of 'java.io.IOException'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)